### DISCUSSION TITLE
[FEATURE] Interactive Live Video Feature for Streamsync

### DISCUSSION THREAD

**Discussion by SurajRaika** [writer/writer-framework#129] [Ideas]
# Proposal: Interactive Live Video Feature for Streamsync

## Introduction
 I'd like to propose the addition of an interactive live video feature, similar to the one found in Streamlit-webrtc.

## Rationale
The motivation behind this proposal is to enhance Streamsync's capabilities by allowing users to modify input values and witness live changes on the video, much like interacting with graphs in Streamlit. 

In my previous work with computer vision applications, I've found that integrating basic UX elements such as buttons or toggle features can be quite challenging. However, the incorporation of this feature could significantly improve our app's user experience.

## Proposal Details
This feature might necessitate some fundamental changes in Streamsync, especially in terms of how the backend communicates with live video on the frontend.

**Comment by ramedina86** [comment]
Hi, I'm familiar with Streamlit-webrtc and Yuichiro Tachibana's great work.

I think WebRTC on Streamsync would be interesting. I don't know how this would be implemented or whether meaningful changes to the backend would be necessary, to be honest I just don't have that kind of knowledge.

The thing that put me off with Streamlit-webrtc is the complexity of the deployment with TURN servers needed for deploying reliably (which cost money!).

Streamsync is quite fast and I've done live video apps before without WebRTC and all the complexity and cost it introduces, just using the Webcam component. Perhaps this can be used instead, maybe with some additional optimisation?

https://github.com/streamsync-cloud/streamsync/assets/64783088/21d65ca9-8555-4b10-98a7-5b58ed9dd128
